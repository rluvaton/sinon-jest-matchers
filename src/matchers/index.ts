export { sinonToBeCalled } from './sinonToBeCalled';
export { sinonToBeCalledTimes } from './sinonToBeCalledTimes';
export { sinonToBeCalledWith } from './sinonToBeCalledWith';
export { sinonLastCalledWith } from './sinonLastCalledWith';
export { sinonLastReturnedWith } from './sinonLastReturnedWith';
export { sinonNthCalledWith } from './sinonNthCalledWith';
export { sinonToReturn } from './sinonToReturn';
export { sinonToReturnTimes } from './sinonToReturnTimes';
export { sinonToReturnWith } from './sinonToReturnWith';
