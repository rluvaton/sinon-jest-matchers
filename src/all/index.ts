import * as matchers from '../matchers';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
const jestExpect = global.expect;

if (jestExpect !== undefined) {
  jestExpect.extend(matchers);
} else {
  throw new Error(
    "Unable to find Jest's global expect. " +
      'Please check you have added sinon-jest-matchers correctly to your jest configuration. ' +
      'See https://github.com/rluvaton/sinon-jest-matchers#setup for help.',
  );
}
