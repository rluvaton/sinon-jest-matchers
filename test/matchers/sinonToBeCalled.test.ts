import sinon from 'sinon';
import * as matchers from '../../src';

expect.extend(matchers);

describe('sinonToBeCalled', () => {
  test('works only on spies or jest.fn', () => {
    const fn = function fn() {
      // noop
    };

    expect(() => expect(fn).sinonToBeCalled()).toThrowErrorMatchingSnapshot();
  });

  test('passes when called', () => {
    const fn = sinon.spy();
    fn('arg0', 'arg1', 'arg2');
    expect(fn).sinonToBeCalled();
    expect(() => expect(fn).not.sinonToBeCalled()).toThrowErrorMatchingSnapshot();
  });

  test('.not passes when called', () => {
    const fn = sinon.spy();

    expect(fn).not.sinonToBeCalled();
    expect(() => expect(fn).sinonToBeCalled()).toThrowErrorMatchingSnapshot();
  });

  test('fails with any argument passed', () => {
    const fn = sinon.spy();

    fn();

    // @ts-expect-error should not get arguments
    expect(() => expect(fn).sinonToBeCalled(555)).toThrowErrorMatchingSnapshot();
  });

  test('.not fails with any argument passed', () => {
    const fn = sinon.spy();

    // @ts-expect-error should not get arguments
    expect(() => expect(fn).not.sinonToBeCalled(555)).toThrowErrorMatchingSnapshot();
  });

  test('includes the custom mock name in the error message', () => {
    const fn = sinon.spy();

    fn.named('named-mock');

    fn();
    expect(fn).sinonToBeCalled();
    expect(() => expect(fn).not.sinonToBeCalled()).toThrowErrorMatchingSnapshot();
  });
});
