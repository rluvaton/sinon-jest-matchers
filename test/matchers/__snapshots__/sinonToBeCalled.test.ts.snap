// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sinonToBeCalled .not fails with any argument passed 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22mnot[2m.[22msinonToBeCalled[2m()[22m

[1mMatcher error[22m: this matcher must not have an expected argument

Expected has type:  number
Expected has value: [32m555[39m"
`;

exports[`sinonToBeCalled .not passes when called 1`] = `
"[2mexpect.[22msinonToBeCalled[2m()[22m

Expected number of calls: >= [32m1[39m
Received number of calls:    [31m0[39m"
`;

exports[`sinonToBeCalled fails with any argument passed 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalled[2m()[22m

[1mMatcher error[22m: this matcher must not have an expected argument

Expected has type:  number
Expected has value: [32m555[39m"
`;

exports[`sinonToBeCalled includes the custom mock name in the error message 1`] = `
"[2mexpect([22m[31mnamed-mock[39m[2m).[22mnot[2m.[22msinonToBeCalled[2m()[22m

Expected number of calls: [32m0[39m
Received number of calls: [31m1[39m

1: called with 0 arguments"
`;

exports[`sinonToBeCalled passes when called 1`] = `
"[2mexpect.[22mnot[2m.[22msinonToBeCalled[2m()[22m

Expected number of calls: [32m0[39m
Received number of calls: [31m1[39m

1: [31m\\"arg0\\"[39m, [31m\\"arg1\\"[39m, [31m\\"arg2\\"[39m"
`;

exports[`sinonToBeCalled works only on spies or jest.fn 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).[22msinonToBeCalled[2m()[22m

[1mMatcher error[22m: [31mreceived[39m value must be a stub or spy function

Received has type:  function
Received has value: [31m[Function fn][39m"
`;
